@navikt/core/react/src/util/focus-boundary/FocusBoundary.tsx

```diff
@@ -0,0 +1,425 @@
+         */
+        if (relatedTarget === null) {
+          return;
+        }
+
+        /*
+         * If the focus has moved to an legitimate element outside the container, we move focus to the last valid focused element inside.
+         * This makes sure to "trap" focus inside the container.
+         */
+        if (!container.contains(relatedTarget)) {
+          focus(lastFocusedElementRef.current, { select: true });
+        }
+      }
+
+      /**
+       * When the currently focused element is removed from the DOM, browsers move focus
+       * to the document.body. In this case, we move focus to the container
+       * to keep focus trapped correctly instead.
+       */
+      const handleMutations = (mutations: MutationRecord[]) => {
+        if (document.activeElement !== document.body) {
+          return;
+        }
+
+        if (mutations.some((mutation) => mutation.removedNodes.length > 0)) {
+          focus(container);
+        }
+      };
+
+      document.addEventListener("focusin", handleFocusIn);
+      document.addEventListener("focusout", handleFocusOut);
+      const observer = new MutationObserver(handleMutations);
+      observer.observe(container, { childList: true, subtree: true });
+
+      return () => {
+        document.removeEventListener("focusin", handleFocusIn);
+        document.removeEventListener("focusout", handleFocusOut);
+        observer.disconnect();
+      };
+    }, [trapped, container, focusBoundary.paused]);
+
+    /* Handles autofocus on mount and unmount */
+    useEffect(() => {
+      if (!container) {
+        return;
+      }
+
+      focusBoundarysStack.add(focusBoundary);
+      const currentActiveElement = document.activeElement as HTMLElement | null;
+      const containsActiveElement =
+        currentActiveElement && container.contains(currentActiveElement);
+
+      /* We only autofocus on mount if container does not contain active element */
+      if (!containsActiveElement) {
+        const mountEvent = new CustomEvent(AUTOFOCUS_ON_MOUNT, EVENT_OPTIONS);
+        container.addEventListener(AUTOFOCUS_ON_MOUNT, onMountAutoFocus);
+        container.dispatchEvent(mountEvent);
+
+        /* If user does not manually prevent event and handle focus themselves */
+        if (!mountEvent.defaultPrevented) {
+          /**
+           * Attempts focusing the first element in a list of candidates.
+           * Stops when focus has actually moved.
+           */
+          const candidates = removeLinks(getTabbableCandidates(container));
+          const previouslyFocusedElement = document.activeElement;
+          for (const candidate of candidates) {
+            focus(candidate, { select: true });
+            if (document.activeElement !== previouslyFocusedElement) {
+              return;
+            }
+          }
+
+          /* focusFirst might not find any candidates, so we fall back to focusing container */
+          if (document.activeElement === currentActiveElement) {
```

Comment:
Should we use `previouslyFocusedElement` here?
```suggestion
          if (document.activeElement === previouslyFocusedElement) {
```

Reply:
Since consumer might end up changing focus here in `onMountAutoFocus`
```
container.addEventListener(AUTOFOCUS_ON_MOUNT, onMountAutoFocus);
container.dispatchEvent(mountEvent);
```
leading to currentActiveElement and previouslyFocusedElement not being the same element, it might be safer to base it on `currentActiveElement`